final-project/application.py

```diff
@@ -143,8 +143,11 @@ def new():
         description=request.form.get("description")
         image=request.form.get("image")
         brand=request.form.get("brand")
-        price=request.form.get("price")
+        priceRaw=request.form.get("price")
 
+        # Stip dollar symbols from the price.
+        if "$" in priceRaw:
```

Comment:
Why is the price stored as a string? You can also use currency codes for simplicity/accuracy.

Reply:
If I saved it as "NUMERIC" I thought it would strip away the trailing zero in something like 8.50,

But I can look into better sqlite3 currency storage.

Reply:
If it's a number, it should be stored as such, at least as a 2 decimal float. Saving currency as strings can be dangerous later on if you actually want to compute on it.

Reply:
![screen shot 2018-03-07 at 8 13 29 am](https://user-images.githubusercontent.com/8930410/37094075-9efadee2-21df-11e8-8e3f-1fa240933225.png)

I send the price to the database as a string to avoid losing the trailing zero. However, the price is actually stored as in the REAL storage class. Sqlite converts the string to a REAL class because I created with price column with a NUMERIC data affinity. The NUMERIC affinity means the string will be converted into either an INTEGER or REAL storage class, depending on what SQLite determines is best.

In this case, it converts the the 8.50 string into the REAL class because of the two decimal places. When a REAL number is returned from SQLite, it is returned as a float (see the screen shot). I did not account for this data flow in my code. In this case the trailing zero is removed. 

I can adjust for my lapse, but I should have stored the price as an integer without any zeros, then added the zeros back on the server. https://dba.stackexchange.com/questions/15729/storing-prices-in-sqlite-what-data-type-to-use

The only use cases this does not account for are the other foreign currencies which use 3-4 significant figures (and cryptocurrencies which use many many signfiicant figures). However, this is all pre-mature optimization.

I understand how I should store currencies in the future. For now, I am going to do the fastest solution and move on. 

Good spotting the error. I learned a lot about currency storage from my reading.

Reply:
Tagged @hdavidzhu 

In case you do not get notifications about the comments, which I do not tag you in.

Reply:
(not relevant for project scope, but for fun:)

Storing currency is actually a pretty interesting software problem. Quick reading on the topic I found so far:

https://stackoverflow.com/questions/3730019/why-not-use-double-or-float-to-represent-currency